soh/soh/Enhancements/randomizer/3drando/fill.cpp

```diff
@@ -118,7 +116,8 @@ static void ValidateWorldChecks(SearchMode& mode, bool checkPoeCollectorAccess,
       std::vector<RandomizerGet> itemsToPlace =
           FilterFromPool(ItemPool, [](const auto i) { return Rando::StaticData::RetrieveItem(i).IsAdvancement(); });
     for (RandomizerGet unplacedItem : itemsToPlace) {
-      Rando::StaticData::RetrieveItem(unplacedItem).ApplyEffect();
+       Rando::StaticData::RetrieveItem(unplacedItem).ApplyEffect();
+       logic->UpdateHelpers();
```

Comment:
Any reason why the `UpdateHelpers` calls were moved?
I think that passing logic as a parameter to `ApplyEffect`/`UndoEffect` and calling `UpdateHelpers` inside of them would be better as that way it's impossible to forget to call `UpdateHelpers` after `ApplyEffect` or `UndoEffect`.

Reply:
I think this was leftover from my first attempts at getting SaveContext relevant, just like the shared_ptr.

Reply:
Wait, I'm not sure I know where they were supposed to be. Someone moved them before, and this was done in a merge conflict resolution.

Reply:
Figured it out.